### PR TITLE
Add presto-prometheus connector

### PR DESCRIPTION
**Pull request by chancez**
This rebases the prometheus_connector branch ontop of master and also adds a few commits:

- presto-prometheus: Remove usage of parseTypeSignature
- Updates to Dockerfile to fix some CI issues (PR to master being made for this).

### CODE REVIEW

**Comment by chancez** [comment]
/hold
I want to make sure this gets reviewed and that all changes besides connector changes make it to master.

File: presto-prometheus/src/main/java/io/prestosql/plugin/prometheus/PrometheusClient.java [hunk 1]
```diff
@@ -82,10 +83,10 @@ public PrometheusClient(PrometheusConnectorConfig config, JsonCodec<Map<String,
 
         tableSupplier = Suppliers.memoizeWithExpiration(metricsSupplier(metricCodec, getPrometheusMetricsURI(config)),
                 timeUnitsToSeconds(config.getCacheDuration()), TimeUnit.SECONDS);
+        MapType varcharMapType = (MapType) typeManager.getType(mapType(VARCHAR.getTypeSignature(), VARCHAR.getTypeSignature()));
```
Comment (bentito) [inline presto-prometheus/src/main/java/io/prestosql/plugin/prometheus/PrometheusClient.java]: yes, this is much more compiler checkable, can see why they targeted this for fixing.

File: presto-prometheus/src/test/java/io/prestosql/plugin/prometheus/TestPrometheusRecordSet.java [hunk 2]
```diff
@@ -51,7 +52,7 @@
     private URI dataUri;
     private static final Metadata METADATA = createTestMetadataManager();
     public static final TypeManager TYPE_MANAGER = new InternalTypeManager(METADATA);
-    static final Type varcharMapType = TYPE_MANAGER.getType(TypeSignature.parseTypeSignature("map(varchar, varchar)"));
+    static final Type varcharMapType = TYPE_MANAGER.getType(mapType(VARCHAR.getTypeSignature(), VARCHAR.getTypeSignature()));
```
Comment (bentito) [inline presto-prometheus/src/test/java/io/prestosql/plugin/prometheus/TestPrometheusRecordSet.java]: Why not get this from MetadataUtil like in `TestPrometheusMetadata.java`?

Comment (chancez) [reply presto-prometheus/src/test/java/io/prestosql/plugin/prometheus/TestPrometheusRecordSet.java]: I just started with this and then did the metadatautil changes after, so I may have missed a few I think. I'll double check.

**Pull request merged**